File: rust/387.first-unique-character-in-a-string.rs
/*
 * @lc app=leetcode id=387 lang=rust
 *
 * [387] First Unique Character in a String
 */

// @lc code=start
impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut chars = vec![0; 26];
        for c in s.bytes() {
            chars[(c - b'a') as usize] += 1;
        }
        if let Some(u) = s.bytes().position(|c| chars[(c - b'a') as usize] == 1) {
            return u as i32;
        }
        return -1;
    }
}
// @lc code=end

File: javascript/387.first-unique-character-in-a-string.js
/*
 * @lc app=leetcode id=387 lang=javascript
 *
 * [387] First Unique Character in a String
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function (s) {
  const chars = new Array(26).fill(0);
  const d = "a".charCodeAt();
  for (const c of s) chars[c.charCodeAt() - d]++;
  for (let i = 0; i < s.length; i++) {
    if (chars[s[i].charCodeAt() - d] === 1) return i;
  }
  return -1;
};
// @lc code=end
